packages/lesswrong/components/async/localStorageHandlers.js

```diff
@@ -0,0 +1,66 @@
+import ls from 'local-storage';
+
+export const commentLSHandlers = {
+  get: ({doc, name}) => {
+    if (doc && doc._id) { // When restoring the edit state for a specific comment, ask for permission
+      const savedState = ls.get(doc._id);
+      if (savedState && Meteor.isClient && window) {
+        const result = window.confirm("We've found a previously saved state for this comment, would you like to restore it?")
+        if (result) { return savedState }
+      } else {
+        return null;
+      }
+    } else if (doc && doc.parentCommentId) {
+      return ls.get('parent:' +  doc.parentCommentId)
+    } else if (doc && doc.postId) {
+      return ls.get('post:' + doc.postId)
+    } else {
+      return null
+    }
+  },
+  set: ({state, doc, name}) => {
```

Comment:
`get`, `set`, and `reset` (in both of the two LS handlers) are sharing a mapping from objects to localstorage IDs, which could be refactored into a method.